packages/io/3mf-serializer/src/index.js

```diff
@@ -0,0 +1,256 @@
+/*
+JSCAD Object to 3MF (XML) Format Serialization
+
+## License
+
+Copyright (c) 2022 JSCAD Organization https://github.com/jscad
+
+All code released under MIT license
+
+Notes:
+1) geom2 conversion to:
+     none
+2) geom3 conversion to:
+     mesh
+3) path2 conversion to:
+     none
+*/
+
+/**
+ * Serializer of JSCAD geometries to 3D manufacturing format (XML)
+ *
+ * The serialization of the following geometries are possible.
+ * - serialization of 3D geometry (geom3) to 3MF object (a unique mesh containing both vertices and triangles)
+ *
+ * Colors are added to base materials when found on the 3D geometry, i.e. attribute color.
+ * Names are added to meshs when found on the 3D geometry, i.e. attribute name.
+ *
+ * @module io/3mf-serializer
+ * @example
+ * const { serializer, mimeType } = require('@jscad/3mf-serializer')
+ */
+
+
+const zipSync = require('fflate').zipSync
+const strToU8 = require('fflate').strToU8
+
+const stringify = require('onml/lib/stringify')
+
+const { colors, geometries, modifiers } = require('@jscad/modeling')
+const { flatten, toArray } = require('@jscad/array-utils')
+
+
+const mimeType = 'model/3mf'
+const fileExtension = '3mf'
+
+/**
+ * Serialize the give objects to 3MF contents (XML) or 3MF packaging (OPC).
+ * @see http://www.3mf.io/specification/
+ * @param {Object} [options] - options for serialization
+ * @param {String} [options.unit='millimeter'] - unit of design; millimeter, inch, feet, meter or micrometer
+ * @param {Boolean} [options.metadata=true] - add metadata to 3MF contents, such at CreationDate
+ * @param {Array} [options.defaultcolor=[0,0,0,1]] - default color for objects
+ * @param {Boolean} [options.compress=true] - package and compress the contents
+ * @param {Object|Array} objects - objects to serialize into 3D manufacturing format
+ * @returns {Array} serialized contents, 3MF contents (XML) or 3MF packaging (ZIP)
+ * @example
+ * const geometry = primitives.cube()
+ * const package = serializer({unit: 'meter'}, geometry) // 3MF package, ZIP format
+ */
+const serialize = (options, ...objects) => {
+  const defaults = {
+    unit: 'millimeter', // micron, millimeter, centimeter, inch, foot, meter
+    metadata: true,
+    defaultcolor: [0, 0, 0, 1], // BLACK
```

Comment:
```suggestion
    defaultcolor: [0, 0.6, 1, 1], // blue
```

Thoughts on using blue instead of black as the default? It would match the jscad style more closely. I'm sure it depends on the viewer, but some viewers don't show pure black very well:

![def-black](https://user-images.githubusercontent.com/1766297/159347206-2a5d7d9c-56f5-43e0-8ec7-556229fafe96.png)

Versus blue:

![def-blue](https://user-images.githubusercontent.com/1766297/159347242-2ab1f6f6-a209-41e6-886e-90183ad9ef40.png)

Reply:
Hmm... 3MF specification says that 'displaycolor' is required, so RGBA has to be set. I'm not a fan of black but it does follow SVG. Hmm...

Reply:
I changed to the default color to JSCAD orange